models/ed/R/veg2model.ED2.R

```diff
@@ -0,0 +1,234 @@
+    soil.dat        <- as.data.frame(matrix(soil, n.patch, 7, byrow = TRUE))
+    names(soil.dat) <- c("fsc", "stsc", "stsl", "ssc", "psc", "msn", "fsn")
+    obs             <- cbind(obs, soil.dat)
+    
+    # Locally write files
+    write.table(obs, localfile, quote = FALSE, col.names = TRUE, row.names = FALSE)
+
+    
+  }else if(inputinfo$output == "css"){
+    
+    if(!is.null(inputinfo$path)){
+      obs <- read.table(inputinfo$path, header = TRUE)
+    }else if(is.null(obs)){
+      logger.severe("No css data found.")
+    }
+    
+    if(inputinfo$source == "FIA"){ # we still need one more check from pss
+      
+      pss <- read.table(runinfo$inputs$pss$path, header = TRUE)
+      
+      # Remove rows that don't map to any patch
+      obs <- obs[which(obs$patch %in% pss$patch), ]
+      if (nrow(obs) == 0) {
+        logger.severe("No trees map to previously selected patches.")
+      } else {
+        logger.debug(paste0(nrow(obs), " trees that map to previously selected patches."))
+      }
+    }
+    
+    if(is.null(obs$n)){
+      # is this how n is calculated?
+      obs$n <- ifelse(!is.null(inputinfo$area), nrow(obs)/inputinfo$area, 0.001)
+    }
+    
+    if(inputinfo$source == "GapMacro"){
+      inv.years <- as.numeric(unique(obs$year))
+      # suitable years
+      av.years <- inv.years[inv.years < start_year]
+      if(length(av.years) == 0){
+        logger.severe("No available years found in the data.")
+      }
+      obs$time <- max(av.years)
+      # filter out other years
+      obs <- obs[obs$year == obs$time, ]
+      colnames(obs)[colnames(obs) == "DBH"] <- "dbh"
+      
+      obs$patch  <- 1
+      obs$cohort <- 1:nrow(obs)
+      obs$hite   <- 0
+    
+    }
+
+    # Convert PFT names to ED2 Numbers
+    data(pftmapping)
+    obs$pft.number <- NA
+    for (p in seq_along(obs$pft)) {
+      obs$pft.number[p] <- pftmapping$ED[pftmapping$PEcAn == as.character(obs$pft[p])]
+      if (is.null(obs$pft.number[p])) {
+        logger.severe(paste0("Couldn't find an ED2 PFT number for ", as.character(obs$pft[p])))
+      }
+    }
+  
+    
+    # --- Continue work formatting css 
+    n.cohort                      <- nrow(obs)
+    obs$time[is.na(obs$time)]     <- 1
+    obs$cohort[is.na(obs$cohort)] <- 1:sum(is.na(obs$cohort))
+    obs$dbh[is.na(obs$dbh)]       <- 1  # assign nominal small dbh to missing
+    density.median                <- median(obs$n[which(obs$n > 0)])
+    obs$n[is.na(obs$n) | obs$n == 0]    <- density.median
+    obs$hite <- obs$bdead <- obs$balive <- obs$lai <- 0
+    
+    # pft.number col needs to be pft 
+    obs <- obs[ , colnames(obs) != "pft"]
+    colnames(obs)[colnames(obs) == "pft.number"] <- "pft"
+    
+    obs <- obs[, c("time", "patch", "cohort", "dbh", "hite", "pft", "n", "bdead", "balive", "lai")]
+    
+    # Locally write files
+    write.table(obs, localfile, quote = FALSE, col.names = TRUE, row.names = FALSE)
+  }
+
+
+  # Insert into DB
+  ic.id <- dbfile.input.insert(
```

Comment:
OK I'll do that, DB insertions were just for testing and was confusing me anyway, I'll get rid of those